pandas/core/generic.py

```diff
@@ -3450,6 +3450,17 @@ def to_csv(
         ...                         archive_name='out.csv')  # doctest: +SKIP
         >>> df.to_csv('out.zip', index=False,
         ...           compression=compression_opts)  # doctest: +SKIP
+
+        Create 'out.csv' in a new folder 'folder/subfolder/'
+
+        >>> import os
+        >>> os.mkdir('folder/subfolder', parents=True, exists_ok=True)
+        >>> df.to_csv('out.csv')
+
+        >>> from pathlib import Path
+        >>> filepath = Path('folder/subfolder/out.csv')
```

Comment:
Do we really need a folder and a subfolder to illustrate the example? Can it just be `folder/out.csv`

Reply:
Could be, yes, but then users would be puzzled when having to use nested folders. This method also works for just `folder/out.csv` and with any level of sub-foldering

Reply:
Please don't link python 2 docs, its unsupported by Python.org and by pandas